### PR TITLE
Align total count on connection & collection segment

### LINKED ISSUE #4955: ConnectionPageInfo.TotalCount not being used at all for Connection's count calculation?
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I can see why there's an option of passing a `Func<CancellationToken, ValueTask>` to get the count, but I don't see why it should be mandatory, especially when the page info object offers a field for count?

Furthermore, I feel like the delegate for counting can be improved. Right now, you can't do anything with it, unless you introduce a closure, which is unnecessary memory overhead. At minimum, I feel like consumers should have access to the list of nodes. Then they can simply do `.Count()` on them to get the count, if that would serve their purpose.

### Steps to reproduce

1. Return a Connection<> object whose `ConnectionPageInfo` object passed has defined `TotalCount` property.
2. The total count used by `Connection<>` does not make use of the one defined in `ConnecetionPageInfo`.

### Relevant log output

_No response_

### Additional Context?

_No response_

### Product

Hot Chocolate

### Version

12.0.0

### PR DESCRIPTION
**Pull request by PascalSenn**
Closes #4955 

- Removes TotalCount from Pageinfo and moves it to `CollectionSegement` and `Connection`.
- Allows to pass the totalCount without a delegate

### CODE REVIEW

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -327,7 +327,7 @@ public async Task Attribute_Simple_StringList_Global_DefaultItem_2()
                 await new ServiceCollection()
                     .AddGraphQL()
                     .AddQueryType<QueryAttr>()
-                    .SetPagingOptions(new PagingOptions { DefaultPageSize = 2 })
+                    .SetPagingOptions(new PagingOptions {DefaultPageSize = 2})
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:330]: until now we always had a space on the braces.

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs [hunk 2]
```diff
@@ -656,44 +678,29 @@ protected override void Configure(IObjectTypeDescriptor<ExecutableQuery> descrip
...
-                "d",
-                "e",
-                "f",
-                "g",
-                "h",
-                "i",
-                "j",
-                "k",
-                "l"
+                "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l"
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:689]: Does this belong in one line now that its reformatted?

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs [hunk 3]
```diff
@@ -718,7 +725,7 @@ public class FluentPaging
                 int? skip,
                 int? take,
                 CancellationToken cancellationToken)
-                => await new[] { "a", "b", "c", "d" }
+                => await new[] {"a", "b", "c", "d"}
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:728]: spaces

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs [hunk 4]
```diff
@@ -751,11 +747,11 @@ public class QueryAttr
                 IncludeTotalCount = true)]
             public List<List<Foo>> Foos() => new List<List<Foo>>
             {
-                new List<Foo> { new Foo { Bar = "a" } },
-                new List<Foo> { new Foo { Bar = "b" }, new Foo { Bar = "c" } },
-                new List<Foo> { new Foo { Bar = "d" } },
-                new List<Foo> { new Foo { Bar = "e" } },
-                new List<Foo> { new Foo { Bar = "f" } }
+                new List<Foo> {new Foo {Bar = "a"}},
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:750]: spaces

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs [hunk 5]
```diff
@@ -751,11 +747,11 @@ public class QueryAttr
...
+                new List<Foo> {new Foo {Bar = "b"}, new Foo {Bar = "c"}},
+                new List<Foo> {new Foo {Bar = "d"}},
```
Comment (michaelstaib) [inline src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:752]: spaces

File: src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs (hunk 2 above)
Comment (PascalSenn) [reply src/HotChocolate/Core/test/Types.OffsetPagination.Tests/IntegrationTests.cs:689]: I just use "chop if long or multine"

**Pull request merged**